components/automate-ui/src/app/components/notification/notification.component.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { async, ComponentFixture, TestBed, fakeAsync, flush } from '@angular/core/testing';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MockComponent } from 'ng2-mock-component';
+import { using } from 'app/testing/spec-helpers';
+
+import { ChefNotificationComponent } from './notification.component';
+import { Type } from 'app/entities/notifications/notification.model';
+
+describe('ChefNotificationComponent', () => {
+  let component: ChefNotificationComponent;
+  let fixture: ComponentFixture<ChefNotificationComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ BrowserAnimationsModule ],
+      declarations: [
+        ChefNotificationComponent,
+        MockComponent({ selector: 'chef-icon' })
+      ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ChefNotificationComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  const genNotification = (type: Type, timeout: number) => {
+    component.type = type;
+    component.timeout = timeout;
+  };
+
+  describe('generates a notification', () => {
+
+    it('and defaults to type of "info" and timeout of 8', () => {
+      expect(component.type).toBe('info');
+      expect(component.timeout).toEqual(8);
+    });
+
+    using([
+      [Type.info, 7],
+      [Type.info, 0],
+      [Type.info, -1],
+      [Type.error, 43],
+      [Type.error, 0]
+    ], function (notificationType: Type, timeLimit: number) {
+        it(`and reads type of ${notificationType} and timeout of ${timeLimit} from inputs`, () => {
+            genNotification(notificationType, timeLimit);
+
+            expect(component.type).toBe(notificationType);
+            expect(component.timeout).toEqual(timeLimit);
+          });
+        });
+  });
+
+  describe('After a notification is visible', () => {
+
+    using([
+      ['is not infinite', Type.info, 10, false],
+      ['is infinite', Type.info, 0, true],
+      ['is not infinite', Type.error, 43, false],
+      ['is infinite', Type.error, 0, true],
+      ['is infinite', Type.error, -1, true]
+    ], function (description: string, type: Type, timeLimit: number, expected: boolean) {
+      it(`${description} when type is ${type} and timeout is ${timeLimit}`, () => {
```

Comment:
Was not quite sure how to interpret what this test is testing--here I just plugged in a set of values:
```
After a notification is visible,
is not infinite when type is Type.info and timeout is 10
```
Seems like you might mean:
```
After a notification appears,
it is scheduled to close in 10 seconds when type is Type.info
```
or
```
After a notification appears,
it remains permanently open when type is Type.info and timeout is -1
```
but those seem to be sufficiently covered in the next 2 sets of tests. 🤷 

(And minor point, but making `this.infinite` public for the sole reason that a test needs it is a little red flag, too.)

Reply:
Good observation - I'll clean that up.

Yeah - I was a little unsure on what to do regarding `this.isinfinite`.  I read that If a variable or function are private, its the responsibility of the test of the function that contains it - but felt safer to test it specifically.  I can pull that out.

Reply:
BTW, if removing the test, then `isinfinite` can be restricted down to a local variable within the function that is using it.